File: packages/point/python_std/point_std/text.py
from __future__ import annotations


def textLength(value: str) -> int:
	return len(value)


def textContains(value: str, search: str) -> bool:
	return search in value


def textSplit(value: str, separator: str) -> list[str]:
	return value.split(separator)


def textTrim(value: str) -> str:
	return value.strip()


def textFromInt(value: int) -> str:
	return str(value)


def textPadStart(value: str, length: int, fill: str) -> str:
	pad_char = fill[:1] if fill else " "
	return value.rjust(length, pad_char)

File: packages/point/src/std/text.ts
export function textLength(value: string): number {
	return value.length;
}

export function textContains(value: string, search: string): boolean {
	return value.includes(search);
}

export function textSplit(value: string, separator: string): string[] {
	return value.split(separator);
}

export function textTrim(value: string): string {
	return value.trim();
}

export function textFromInt(value: number): string {
	return String(value);
}

export function textPadStart(value: string, length: number, fill: string): string {
	const padChar = fill.length > 0 ? fill.slice(0, 1) : " ";
	return value.padStart(length, padChar);
}
